File: shorturl-app/src/main/java/pl/mpakula/shorturl/infrastructure/UrlMappingJpaRepository.java
package pl.mpakula.shorturl.infrastructure;


import lombok.NonNull;
import org.springframework.data.repository.Repository;

import java.math.BigInteger;
import java.util.Optional;

interface UrlMappingJpaRepository extends Repository<UrlMappingJpa, BigInteger> {

    UrlMappingJpa save(@NonNull UrlMappingJpa urlMapping);

    Optional<UrlMappingJpa> findFirstByOriginalUrl(String originalUrl);

    Optional<UrlMappingJpa> findById(BigInteger id);

}

File: repository-db-adapter/src/main/java/pl/mpakula/shorturl/infrastructure/UrlMappingJpaRepository.kt
package pl.mpakula.shorturl.infrastructure

import org.springframework.data.repository.Repository
import java.math.BigInteger

internal interface UrlMappingJpaRepository : Repository<UrlMappingJpa, BigInteger> {
    fun save(urlMapping: UrlMappingJpa): UrlMappingJpa
    fun findFirstByOriginalUrl(originalUrl: String?): UrlMappingJpa?
    fun findById(id: BigInteger?): UrlMappingJpa?
}